warnings on mac

Introduced by fixing #79 

Warning logs on building:

```make
/Users/adit/projects/expresso/example/main.cpp:73:11: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
   73 |   app.use(move(cors));
      |           ^
      |           std::
/Users/adit/projects/expresso/example/main.cpp:78:11: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
   78 |   app.use(move(cookieParser));
      |           ^
      |           std::
/Users/adit/projects/expresso/example/main.cpp:83:11: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
   83 |   app.use(move(staticServe));
      |           ^
      |           std::
/Users/adit/projects/expresso/example/main.cpp:99:21: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
   99 |   pictureRouter.use(move(pictureServe));
      |                     ^
      |                     std::
In file included from /Users/adit/projects/expresso/example/main.cpp:2:
In file included from /Users/adit/projects/expresso/include/expresso/core/server.h:6:
In file included from /Users/adit/projects/expresso/lib/json/include/json/parse.h:4:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/sstream:278:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/istream:170:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/ostream:179:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/shared_ptr.h:33:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:66:5: warning: delete called on 'expresso::middleware::Middleware' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
   66 |     delete __ptr;
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:266:7: note: in instantiation of member function 'std::default_delete<expresso::middleware::Middleware>::operator()' requested here
  266 |       __ptr_.second()(__tmp);
      |       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:236:71: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::reset' requested here
  236 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
/Users/adit/projects/expresso/example/main.cpp:73:11: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::~unique_ptr' requested here
   73 |   app.use(move(cors));
      |           ^
5 warnings generated.
[  6%] Building CXX object CMakeFiles/server.dir/src/core/cookie.cpp.o
[  9%] Building CXX object CMakeFiles/server.dir/src/core/request.cpp.o
[ 12%] Building CXX object CMakeFiles/server.dir/src/core/response.cpp.o
[ 16%] Building CXX object CMakeFiles/server.dir/src/core/router.cpp.o
In file included from /Users/adit/projects/expresso/src/core/router.cpp:1:
In file included from /Users/adit/projects/expresso/include/expresso/core/router.h:3:
In file included from /Users/adit/projects/expresso/include/expresso/middleware/middleware.h:3:
In file included from /Users/adit/projects/expresso/include/expresso/core/request.h:3:
In file included from /Users/adit/projects/expresso/include/expresso/core/response.h:4:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/sstream:278:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/istream:170:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/ostream:179:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/shared_ptr.h:33:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:66:5: warning: delete called on 'expresso::middleware::Middleware' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
   66 |     delete __ptr;
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:266:7: note: in instantiation of member function 'std::default_delete<expresso::middleware::Middleware>::operator()' requested here
  266 |       __ptr_.second()(__tmp);
      |       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:236:71: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::reset' requested here
  236 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/allocator.h:176:87: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::~unique_ptr' requested here
  176 |   _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_HIDE_FROM_ABI void destroy(pointer __p) { __p->~_Tp(); }
      |                                                                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/allocator_traits.h:311:9: note: in instantiation of member function 'std::allocator<std::unique_ptr<expresso::middleware::Middleware>>::destroy' requested here
  311 |     __a.destroy(__p);
      |         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/vector:926:23: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<std::unique_ptr<expresso::middleware::Middleware>>>::destroy<std::unique_ptr<expresso::middleware::Middleware>, void>' requested here
  926 |       __alloc_traits::destroy(__alloc(), std::__to_address(--__soon_to_be_end));
      |                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/vector:920:5: note: in instantiation of member function 'std::vector<std::unique_ptr<expresso::middleware::Middleware>>::__base_destruct_at_end' requested here
  920 |     __base_destruct_at_end(this->__begin_);
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/vector:490:16: note: in instantiation of member function 'std::vector<std::unique_ptr<expresso::middleware::Middleware>>::__clear' requested here
  490 |         __vec_.__clear();
      |                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/vector:501:67: note: in instantiation of member function 'std::vector<std::unique_ptr<expresso::middleware::Middleware>>::__destroy_vector::operator()' requested here
  501 |   _LIBCPP_CONSTEXPR_SINCE_CXX20 _LIBCPP_HIDE_FROM_ABI ~vector() { __destroy_vector (*this)(); }
      |                                                                   ^
/Users/adit/projects/expresso/src/core/router.cpp:3:25: note: in instantiation of member function 'std::vector<std::unique_ptr<expresso::middleware::Middleware>>::~vector' requested here
    3 | expresso::core::Router::Router() : paramRouter(nullptr) { return; }
      |                         ^
1 warning generated.
[ 19%] Building CXX object CMakeFiles/server.dir/src/core/server.cpp.o
In file included from /Users/adit/projects/expresso/src/core/server.cpp:1:
In file included from /Users/adit/projects/expresso/include/expresso/core/server.h:6:
In file included from /Users/adit/projects/expresso/lib/json/include/json/parse.h:4:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/sstream:278:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/istream:170:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/ostream:179:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/shared_ptr.h:33:
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:66:5: warning: delete called on 'expresso::middleware::Middleware' that is abstract but has non-virtual destructor [-Wdelete-abstract-non-virtual-dtor]
   66 |     delete __ptr;
      |     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:266:7: note: in instantiation of member function 'std::default_delete<expresso::middleware::Middleware>::operator()' requested here
  266 |       __ptr_.second()(__tmp);
      |       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/c++/v1/__memory/unique_ptr.h:236:71: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::reset' requested here
  236 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 ~unique_ptr() { reset(); }
      |                                                                       ^
/Users/adit/projects/expresso/src/core/server.cpp:59:13: note: in instantiation of member function 'std::unique_ptr<expresso::middleware::Middleware>::~unique_ptr' requested here
   59 |   this->use(std::make_unique<expresso::middleware::Version>());
      |             ^
1 warning generated.
[ 22%] Building CXX object CMakeFiles/server.dir/src/helpers/response.cpp.o
[ 25%] Building CXX object CMakeFiles/server.dir/src/middleware/cookie_parser.cpp.o
[ 29%] Building CXX object CMakeFiles/server.dir/src/middleware/cors.cpp.o
[ 32%] Building CXX object CMakeFiles/server.dir/src/middleware/date.cpp.o
[ 35%] Building CXX object CMakeFiles/server.dir/src/middleware/middleware.cpp.o
[ 38%] Building CXX object CMakeFiles/server.dir/src/middleware/static_serve.cpp.o
[ 41%] Building CXX object CMakeFiles/server.dir/src/middleware/version.cpp.o
[ 45%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/base64.cpp.o
[ 48%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/date.cpp.o
[ 51%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/env.cpp.o
[ 54%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/os.cpp.o
[ 58%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/os/dir.cpp.o
[ 61%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/os/file.cpp.o
[ 64%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/string.cpp.o
[ 67%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/sys.cpp.o
[ 70%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/url.cpp.o
[ 74%] Building CXX object CMakeFiles/server.dir/lib/brewtils/src/uuid.cpp.o
[ 77%] Building CXX object CMakeFiles/server.dir/lib/json/src/object.cpp.o
[ 80%] Building CXX object CMakeFiles/server.dir/lib/json/src/parse.cpp.o
[ 83%] Building CXX object CMakeFiles/server.dir/lib/logger/src/config.cpp.o
[ 87%] Building CXX object CMakeFiles/server.dir/lib/logger/src/log.cpp.o
[ 90%] Building CXX object CMakeFiles/server.dir/lib/nexus/src/pool.cpp.o
[ 93%] Building CXX object CMakeFiles/server.dir/lib/zippuccino/src/crc.cpp.o
[ 96%] Building CXX object CMakeFiles/server.dir/lib/zippuccino/src/zipper.cpp.o
[100%] Linking CXX executable server
[100%] Built target server
```

--- a/example/main.cpp
+++ b/example/main.cpp
@@ -8,7 +8,6 @@
 
 // Personally, I don't encourange using namespaces, but, I left it here just so
 // that the code could be more readable ¯\_(ツ)_/¯
-using namespace std;
 using namespace expresso::core;
 using namespace expresso::enums;
 using namespace expresso::middleware;
@@ -67,20 +66,20 @@ int main(int argc, char **argv) {
   Server app = Server();
 
   // CORS middleware
-  unique_ptr<expresso::middleware::Cors> cors = make_unique<Cors>();
+  std::unique_ptr<expresso::middleware::Cors> cors = std::make_unique<Cors>();
   cors->allowOrigin("*");
   cors->allowCredentials(true);
-  app.use(move(cors));
+  app.use(std::move(cors));
 
   // Cookie Parser
-  unique_ptr<expresso::middleware::CookieParser> cookieParser =
-      make_unique<CookieParser>();
-  app.use(move(cookieParser));
+  std::unique_ptr<expresso::middleware::CookieParser> cookieParser =
+      std::make_unique<CookieParser>();
+  app.use(std::move(cookieParser));
 
   // Static serve middleware
-  unique_ptr<expresso::middleware::StaticServe> staticServe =
-      make_unique<StaticServe>("../assets");
-  app.use(move(staticServe));
+  std::unique_ptr<expresso::middleware::StaticServe> staticServe =
+      std::make_unique<StaticServe>("../assets");
+  app.use(std::move(staticServe));
 
   // Route handling like normal
   app.get("/health", [](Request &req, Response &res) {
@@ -93,10 +92,10 @@ int main(int argc, char **argv) {
   app.use("/about", &router);
 
   // Listing directories
-  unique_ptr<expresso::middleware::StaticServe> pictureServe =
-      make_unique<StaticServe>("../assets/github", true);
+  std::unique_ptr<expresso::middleware::StaticServe> pictureServe =
+      std::make_unique<StaticServe>("../assets/github", true);
   Router pictureRouter;
-  pictureRouter.use(move(pictureServe));
+  pictureRouter.use(std::move(pictureServe));
   app.use("/pictures", &pictureRouter);
 
   // Sending multiple files as single zip

--- a/include/expresso/middleware/middleware.h
+++ b/include/expresso/middleware/middleware.h
@@ -10,7 +10,7 @@ class Middleware {
 private:
 public:
   Middleware();
-  ~Middleware();
+  virtual ~Middleware() = default;
 
   virtual bool use(expresso::core::Request &req,
                    expresso::core::Response &res) = 0;

--- a/src/middleware/middleware.cpp
+++ b/src/middleware/middleware.cpp
@@ -1,5 +1,3 @@
 #include <expresso/middleware/middleware.h>
 
 expresso::middleware::Middleware::Middleware() { return; }
-
-expresso::middleware::Middleware::~Middleware() { return; }

